drivers/usb/udc/Kconfig.mcux

```diff
@@ -0,0 +1,24 @@
+# Copyright 2024 NXP
+# SPDX-License-Identifier: Apache-2.0
+
+config USB_UDC_NXP_EHCI
+	bool "NXP MCUX USB EHCI Device controller driver"
+	default y
+	depends on DT_HAS_NXP_EHCI_ENABLED
+	select NOCACHE_MEMORY if HAS_MCUX_CACHE && CPU_HAS_DCACHE
+	help
+	  NXP MCUX USB Device Controller Driver for EHCI.
+
+config USB_UDC_NXP_IP3511
+	bool "NXP MCUX USB IP3511 Device controller driver"
+	default y
+	depends on DT_HAS_NXP_LPCIP3511_ENABLED
+	help
+	  NXP MCUX USB Device Controller Driver for KHCI.
+
+config USB_UDC_NXP_PHY
+	bool "NXP MCUX USB Phy"
+	default y
+	depends on DT_HAS_NXP_USBPHY_ENABLED
+	help
+	  NXP MCUX USB Phy.
```

Comment:
I see the only justification for this is use in `modules/hal_nxp/usb/CMakeLists.txt`. Can we use DT_HAS_NXP_USBPHY_ENABLED in `modules/hal_nxp/usb/CMakeLists.txt` there directly?

Reply:
Remove `COFNIG_USB_UDC_NXP_PHY` and use `CONFIG_DT_HAS_NXP_USBPHY_ENABLED `